src/main/java/core/dao/car/CarDaoJdbc.java

```diff
@@ -0,0 +1,215 @@
+        }
+        return updated > 0;
+    }
+    
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> returnList = new ArrayList<>();
+        String getAllByDriver = "SELECT cars.id as car_id, cars.model as car_model,"
+                                + " cars.manufacturer as manufacturer_id,"
+                                + " m.name as manufacturer_name, m.country as manufacturer_country,"
+                                + " d.id as driver_id, d.name as driver_name,"
+                                + " d.licence_number as driver_licence"
+                                + " FROM cars"
+                                + " LEFT JOIN cars_drivers cd ON cd.\"car_Id\" = cars.id"
+                                + " LEFT JOIN drivers d on cd.\"driver_Id\" = d.id"
+                                + " LEFT JOIN manufacturers m ON m.id = cars.manufacturer"
+                                + " WHERE cars.deleted = false AND d.deleted = false"
+                                + " AND cars.id in (select \"car_Id\" FROM cars_drivers "
+                                + " WHERE \"driver_Id\" = ?)"
+                                + " ORDER BY cars.id";
+        try (Connection connection = ConnectionUtils.getConnection();
+                 PreparedStatement getAllStatement = connection.prepareStatement(getAllByDriver)) {
+            getAllStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllStatement.executeQuery();
+            if (resultSet.next()) {
+                returnList = getListCarParser(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(String
+                    .format(GET_CARS_BY_DRIVER_EXCEPTION, driverId), e);
+        }
+        return returnList;
+    }
+    
+    private void insertDrivers(Car car, Connection connection) {
+        String insert = "INSERT INTO cars_drivers(\"car_Id\", \"driver_Id\") VALUES ("
+                        + car.getId() + ", ?);";
+        try (PreparedStatement insertStatement = connection.prepareStatement(insert)) {
+            for (Driver driver : car.getDriverList()) {
+                insertStatement.setLong(1, driver.getId());
+                int wasAdded = insertStatement.executeUpdate();
+                if (wasAdded == 0) {
+                    throw new DataProcessingException(String.format(INSERT_DRIVER_EXCEPTION,
+                            driver.getId()));
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(String.format(INSERT_DRIVER_EXCEPTION,
+                    car.getId()), e);
+        }
+    }
+    
+    private void removeDrivers(Car car, Connection connection) {
+        String remove = "DELETE FROM cars_drivers WHERE \"car_Id\" = " + car.getId()
+                + " AND EXISTS(SELECT id WHERE \"car_Id\" = ?)";
```

Comment:
`" AND EXISTS(SELECT id WHERE \"car_Id\" = ?)";` - what is it for? can we remove it?

Reply:
Sure. Thanks.